python/ray/dag/dag_node.py

```diff
@@ -335,6 +354,33 @@ def __call__(self, node: "DAGNode"):
             )
         )
 
+    def bfs(self, fn: "Callable[[DAGNode], T]"):
```

Comment:
maybe we should consider whether it is possible to use BFS instead of `apply_recursive` for better readability.

Reply:
let's do it in a separate PR!